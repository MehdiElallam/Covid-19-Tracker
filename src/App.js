import './App.css';
import Header from './components/Header';

function App() {
  return (
    <div className="App">
      <Header title="Covid-19 Tracker" />
    </div>
  );
}

export default App;
