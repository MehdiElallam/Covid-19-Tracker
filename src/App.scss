
$main-color : #f8f9fa;
$border-color: rgb(226, 226, 226);

body {
  background-color: $main-color;
  
  .sidebar {
    height : 100vh;
    border-right: 1px solid $border-color;
  }
}


// WE CAN USE NESTING AND MIXING :