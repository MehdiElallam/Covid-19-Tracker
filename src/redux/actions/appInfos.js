import { GET_PAGE_TITLE } from '../types'

export const changePageTitle = (title) => {
    
    return (dispatch) => {
        
        dispatch({
            type : GET_PAGE_TITLE,
            payload : title
        })
    }
    
}