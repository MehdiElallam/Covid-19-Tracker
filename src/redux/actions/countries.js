import {
    GET_WORLD_COUNTRIES_REQUEST, 
    GET_WORLD_COUNTRIES_SUCCESS, 
    GET_WORLD_COUNTRIES_ERROR,
} from '../types';
import {instance } from '../../utils/config'



export const fetchWorldCountriesRequest = () => {
    return {
        type : GET_WORLD_COUNTRIES_REQUEST
    }
}

export const fetchWorldCountriesSuccess = (countries) => {
    return {
        type : GET_WORLD_COUNTRIES_SUCCESS,
        payload : countries
    }
}

export const fetchWorldCountriesError = (error) => {
    return {
        type : GET_WORLD_COUNTRIES_ERROR,
        payload : error
    }
}

export const fetchWorldCountries = () => {
    
    return async (dispatch) => {
        
        dispatch(fetchWorldCountriesRequest)
        try {
            const countries = await instance.get('/countries')
            dispatch(fetchWorldCountriesSuccess(countries.data))
        }
        catch(error){
            const errorMsg = error.message
            dispatch(fetchWorldCountriesError(errorMsg))
        }
    }   
}