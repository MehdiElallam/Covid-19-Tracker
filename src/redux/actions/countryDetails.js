import {
    GET_COUNTRY_DATA_REQUEST, 
    GET_COUNTRY_DATA_SUCCESS, 
    GET_COUNTRY_DATA_ERROR,
    GET_COUNTRY_HISTORICAL_REQUEST,
    GET_COUNTRY_HISTORICAL_SUCCESS,
    GET_COUNTRY_HISTORICAL_ERROR

} from '../types'
import { instance } from '../../utils/config'

export const fetchCountryDataRequest = () => {
    return {
        type : GET_COUNTRY_DATA_REQUEST,
    }
}

export const fetchCountryDataSuccess = (data) => {
    return {
        type : GET_COUNTRY_DATA_SUCCESS,
        payload : data
    }
}

export const fetchCountryDataError = (error) => {
    return {
        type : GET_COUNTRY_DATA_ERROR,
        payload : error
    }
}

export const fetchCountryData = (country) => {
    
    return async(dispatch) => {
        
        dispatch(fetchCountryDataRequest)
        try {
            const countryDetails = await instance.get(`/countries/${country}`)
            dispatch(fetchCountryDataSuccess(countryDetails.data))
        }
        catch(error) {
            
            const errorMsg = error.message
            dispatch(fetchCountryDataError(errorMsg))
        }
    }
    
}

export const fetchCountryHistoricalRequest = () => {
    return {
        type : GET_COUNTRY_HISTORICAL_REQUEST,
    }
}

export const fetchCountryHistoricalSuccess = (data) => {
    return {
        type : GET_COUNTRY_HISTORICAL_SUCCESS,
        payload : data
    }
}

export const fetchCountryHistoricalError = (error) => {
    return {
        type : GET_COUNTRY_HISTORICAL_ERROR,
        payload : error
    }
}

export const fetchCountryHistorical = (country) => {
    
    return async(dispatch) => {
        
        dispatch(fetchCountryHistoricalRequest)
        try {
            const countryHistorical = await instance.get(`/historical/${country}?lastdays=365`)
            dispatch(fetchCountryHistoricalSuccess(countryHistorical.data))
        }
        catch(error) {
            
            const errorMsg = error.message
            dispatch(fetchCountryHistoricalError(errorMsg))
        }
    }
    
}