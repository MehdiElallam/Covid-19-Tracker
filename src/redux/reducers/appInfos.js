import { GET_PAGE_TITLE, DEFAULT_COUNTRY } from '../types'
 
const initialState = {
    infos : {
        title : 'Daily update',
        countryCode : 'UK'
    }
}

export default function appReducer (state = initialState, action){
    switch(action.type){
        case GET_PAGE_TITLE:
            return {
                ...state,
                infos : {
                    ...state.infos,
                    title : action.payload
                }
                
            }

        case DEFAULT_COUNTRY:
                return {
                    ...state,
                    infos : {
                        ...state.infos,
                        countryCode : action.payload
                    }
                    
                }
        
        default: return state
    }

}
