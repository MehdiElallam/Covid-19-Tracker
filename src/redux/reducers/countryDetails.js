import {
    GET_COUNTRY_DATA_REQUEST, 
    GET_COUNTRY_DATA_SUCCESS, 
    GET_COUNTRY_DATA_ERROR,
    GET_COUNTRY_HISTORICAL_REQUEST,
    GET_COUNTRY_HISTORICAL_SUCCESS,
    GET_COUNTRY_HISTORICAL_ERROR

} from '../types'
 
const initialState = {
    countryLoading : true,
    historicalLoading : true,
    country : {},
    historical : {}
}

export default function (state = initialState, action){

    switch(action.type){

        case GET_COUNTRY_DATA_REQUEST:
            return {
                ...state,
                countryLoading : true
            }

        case GET_COUNTRY_DATA_SUCCESS:
            return {
                ...state,
                country : {
                    ...state.country,
                    total : action.payload
                },
                countryLoading : false
            }

        case GET_COUNTRY_DATA_ERROR:
            return {
                ...state,
                error : action.payload,
                country : {
                    ...state.country,
                    total : {}
                },
                countryLoading:false
            }

        case GET_COUNTRY_HISTORICAL_REQUEST:
            return {
                ...state,
                historicalLoading : true
            }

        case GET_COUNTRY_HISTORICAL_SUCCESS:
            return {
                ...state,
                country : {
                    ...state.country,
                    historical : action.payload
                },
                historicalLoading : false
            }

        case GET_COUNTRY_HISTORICAL_ERROR:
            return {
                ...state,
                error : action.payload,
                country : {
                    ...state.country,
                    historical : {}
                },
                historicalLoading:false
            }
        
        default: return state
    }

}
