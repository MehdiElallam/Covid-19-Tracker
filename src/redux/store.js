import {createStore, applyMiddleware, compose} from 'redux'
import thunk from 'redux-thunk'
import rootReducer from './reducers'

const initialState = {}

const middlewares = [thunk]

// const store = createStore(rootReducer, initialState, applyMiddleware(...middlewares),window.devToolsExtension && window.devToolsExtension())

const store = compose(
    applyMiddleware(thunk),
    window.devToolsExtension && window.devToolsExtension(),
)(createStore)(rootReducer, initialState);

export default store;