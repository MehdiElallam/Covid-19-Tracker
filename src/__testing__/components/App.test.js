import React from "react";
import { shallow } from 'enzyme';
import App from "../../../src/App";
import {Provider} from 'react-redux';
import store from "../../redux/store" 
import { changePageTitle } from '../../redux/actions/appInfos'
import { fetchCountryHistorical } from "../../redux/actions/countryDetails"


import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

describe("Rendring App component", () => {
    
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
            historical : {},
        });

        store.dispatch = jest.fn();

        component = shallow(
            <Provider store={store}>
                <App />
            </Provider>   
        );
    });

    it('should dispatch an action on mount', () => {
        store.dispatch(fetchCountryHistorical())
        expect(store.dispatch).toHaveBeenCalledTimes(1);
    });

    it('should render with given state from Redux store', () => {
        expect(component).toMatchSnapshot();
    });

    

});