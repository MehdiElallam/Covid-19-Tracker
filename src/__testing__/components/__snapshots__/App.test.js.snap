// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `ShallowWrapper {}`;

exports[`Rendring App component should render with given state from Redux store 1`] = `ShallowWrapper {}`;

exports[`Rendring the correct content LineGraph should render correctly with no props 1`] = `ShallowWrapper {}`;

exports[`Rendring the correct content renders correctly 1`] = `ShallowWrapper {}`;
